docs/_guide/05-creating-directives.md

```diff
@@ -0,0 +1,221 @@
+
+To create nested parts, you construct [`NodePart`](/api/classes/lit_html.nodepart.html) instances and associate them with specific locations in the DOM. The section of DOM controlled by a given `NodePart` needs to be delimited by static nodes that serve as markers. (lit-html usually uses comment nodes for these markers.)
+
+<img alt="Diagram showing a tree of DOM nodes and a NodePart object. The DOM tree consists of a parent node and seveal child nodes, with two of the child nodes identified as 'marker nodes.' The NodePart object has a startNode property, which points to the first marker node, and an endNode property, which points to the second marker node. Child nodes between the two marker nodes are identified as 'nodes managed by NodePart.'" src="/images/guides/node-part-markers.png" style="max-width: 515px;">
+
+As shown in the diagram, the nodes managed by the `NodePart` appear between its `startNode` and `endNode`. The following code creates and adds a new, nested part inside an existing part (the "container part"). 
+
+```js
+import {NodePart, createMarker} from 'lit-html';
+
+// Create a new part, passing in the render options from the original part
+const newPart = new NodePart(containerPart.options);
+
+// Create markers surrounding content managed by the new part
+const container = containerPart.startNode.parentNode;
+const startNode = createMarker();
+container.insertBefore(startNode, containerPart.endNode);
+container.insertBefore(createMarker(), containerPart.endNode);
+
+newPart.insertAfterNode(startNode);
+```
+
+If you don't need to manage the marker nodes yourself, you can use the [appendIntoPart](/api/classes/lit_html.nodepart.html#appendintopart) convenience method: 
+
+```js
+import {NodePart} from 'lit-html';
+const newPart = new NodePart(containerPart.options);
+
+newPart.appendIntoPart(containerPart);
+```
+
+The end result looks something like this:
+
+<img alt="Diagram showing a tree of DOM nodes and a two NodePart objects. The DOM tree consists of a parent node and seveal child nodes, with two pairs of child nodes identified as 'marker nodes.' The cointainer NodePart object has a startNode property, which points to the first marker node, and an endNode property, which points to the last marker node. The nested NodePart object has startNode and endNode properties that point to the second and third marker nodes. Child nodes between the second and third marker nodes are identified as 'nodes managed by nested NodePart.'" src="/images/guides/nested-node-parts.png" style="max-width: 535px;">
+
+Putting it all together—the following example directive takes a value and inserts it into the DOM _twice_ by creating two nested parts.
+
+```js
+// Import lit-html APIs
+import {html, render, directive, NodePart, createMarker} from 'lit-html';
+
+// duplicate directive takes a single value, and renders it
+// in the DOM twice
+const duplicate = directive((value) => {
+  let part1;
+  let part2;
```

Comment:
Huh. I was afraid I was going to have to use that approach. Although once again I am baffled by JS. I thought in this case the local vars would be inside the closure of the factory function, so would be unique to one instance of the directive. 

The more I learn, the less I know. Anyway, weak maps it is.

Reply:
> The (only) place that the current directive API has to hang state that can persist between renders is the part argument passed into the directive

I've found a catch regarding this behavior: when the directive is rendered inside a nested template, the part state is not persisted after toggling the render of the nested template.

Here's an example: https://codepen.io/blikblum/pen/BbeWXO?editors=1010

Note that when "Toggle" is clicked, the directive part state is kept between toggle states when called directly but not when the directive is inside the nested template

Reply:
@blikblum I think what you're doing in your example is removing and recreating the inner template (or rather, the DOM managed by the inner template) when you click toggle. So that would be working as I expect based on @kevinpschaaf's description, above.

Reply:
I understand. Anyway, should be noted that the ability to nest templates may lead the user to think that directives in nested templates act the same as inlined directives. In fact most of the time does, but not in specific behaviors like this.